catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/services/database/DatabaseServiceResource.java

```diff
@@ -352,9 +360,50 @@ public Response delete(
       throws IOException, ParseException {
     SecurityUtil.checkAdminOrBotRole(authorizer, securityContext);
     DeleteResponse<DatabaseService> response = dao.delete(securityContext.getUserPrincipal().getName(), id, recursive);
+    decrypt(securityContext, response.getEntity());
     return response.toResponse();
   }
 
+  @POST
+  @Path("/rotate")
+  @Operation(
+      summary = "Rotate the fernet key",
+      tags = "services",
+      description =
+          "Once connection credentials have been encrypted using a fernet key, changing the key will cause decryption"
+              + " of existing credentials to fail. To rotate the fernet key without invalidating existing encrypted values,"
+              + " prepend the new key to the fernetConfiguration.fernetKey setting, and then invoke this API",
+      responses = {@ApiResponse(responseCode = "200", description = "OK")})
+  public Response rotate(@Context UriInfo uriInfo, @Context SecurityContext securityContext)
+      throws IOException, ParseException, GeneralSecurityException {
+    SecurityUtil.checkAdminOrBotRole(authorizer, securityContext);
+    dao.rotate();
+    return Response.ok().build();
+  }
+
+  private ResultList<DatabaseService> decrypt(
+      SecurityContext securityContext, ResultList<DatabaseService> databaseServices) {
+    Optional.ofNullable(databaseServices.getData())
+        .orElse(Collections.emptyList())
+        .forEach(databaseService -> decrypt(securityContext, databaseService));
+    return databaseServices;
+  }
+
+  private DatabaseService decrypt(SecurityContext securityContext, DatabaseService databaseService) {
+    try {
+      SecurityUtil.checkAdminRoleOrPermissions(authorizer, securityContext, null, MetadataOperation.DecryptTokens);
```

Comment:
Should we also allow the owner of the database service to decrypt?

Reply:
@harshach 
Here we decrypt if normal user but the owner. In `deploy` airflow pipeline, we don't. Is it ok?

Reply:
I have rolled back the previous version because an entity without an owner is owned by everybody. That means that anybody could see the password if the DatabaseService doesn't have an owner that is most of the cases. Only admins or bots can see the password in clear. In the other cases, we send `null` for the entire DatabaseConnection object.